### PR TITLE
Boost: Add "load default strings" to excludes in concatenate UI

### PR DESCRIPTION
**Pull request by dilirity**
## Proposed changes:

* Add a button, similar to cornerstone pages, that loads default exclude values for concatenate JS and CSS;
* Clean up CSS in minify component.

CleanShot 2024-12-06 at 16 59 15

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

- Setup Boost;
- Make sure concatenate JS/CSS UI is behaving as expected.

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx [hunk 1]
```diff
@@ -35,11 +43,15 @@ const MetaComponent = ( { buttonText, placeholder, datasyncKey }: Props ) => {
...
+	if ( values.length > 0 ) {
+		/* Translators: %s refers to the list of excluded items. */
+		summary = sprintf( __( 'Except: %s', 'jetpack-boost' ), values.join( ', ' ) );
+	}
+
+	if ( values.length === 0 ) {
+		summary = __( 'No exceptions.', 'jetpack-boost' );
+	}
 
 	// Be explicit about this because the optimizer breaks the linter otherwise.
```
Comment (donnchawp) [inline projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: We should probably move this comment to line 46 because summary is being defined in the same way.

File: projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx [hunk 2]
```diff
@@ -69,14 +93,24 @@ const MetaComponent = ( { buttonText, placeholder, datasyncKey }: Props ) => {
...
+					>
+						{ __( 'Save', 'jetpack-boost' ) }
+					</Button>
+					<Button
+						disabled={ inputValue === defaultValue }
+						onClick={ loadDefaultValue }
+						className={ styles.button }
+						variant="link"
+					>
+						{ __( 'Load default strings', 'jetpack-boost' ) }
```
Comment (donnchawp) [inline projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I'd much rather we didn't use "strings" here or in the sub header. Yes, they literally are strings, but they're the handles used to register JS and CSS files. Strings describes what those pixels on the screen are, not what they do or represent
.
Maybe use "scripts" instead? "JS Scripts" "CSS Scripts" ?

I'm not sure. Could be a job for another PR.

**Review by donnchawp — CHANGES_REQUESTED** [review]
Some minor things, but looks good otherwise.

File: projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx (hunk 2 above)
Comment (haqadn) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I agree. The values aren't just strings. They are handles. I'd suggest calling them "default handles" as well.

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I don't mind calling them handles, I'm worried that a lot of users have no idea what that means. We're mentioning strings everywhere.

File: projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx (hunk 1 above)
Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: Updated in https://github.com/Automattic/jetpack/pull/40496/commits/b143e7b2b2870e834b6de9cbc17e6ac2e7b56ab4.

File: projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx (hunk 2 above)
Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: Changed in [`34fde9b`](https://github.com/Automattic/jetpack/pull/40496/commits/34fde9bb9eedd7252c16fc4a3e5593b4d04174a2).

We should probably change the labels as well.

Comment (donnchawp) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I should have looked more carefully at the modules.

In the JS module:
1. "Scripts" is used to describe the files loaded. 
2. The toggle says, "Exclude JS Strings".
3. The sub header says, "Exclude JS Strings".

In the CSS module:
1. "Styles" is used to describe the files loaded.
2. The toggle says, "Exclude CSS Strings".
3. The sub header says, "Exclude CSS Strings".

This patch adds a button saying, "Load default handles".

Would it be better to use "scripts" in the JS module, and "styles" in the CSS module? This is what it might look like:

Screenshot 2024-12-11 at 12 06 48

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: What if we drop "Scripts/Styles" from the label and just say "Exclude JS" or "Exclude CSS".

Then we could update the description to say "handles" and maybe add a link to a support article that mentions how to identify those? Then the button will be "Load default handles".

Comment (donnchawp) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: That makes sense when I say "JS scripts" out loud, as in "JavaScript scripts". It's a bit cumbersome.
I'd go one step further and change that last string to "Load Defaults" too. No need to confuse the user with new jargon.

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: > I'd go one step further and change that last string to "Load Defaults" too.

Funny you should say that. I intentionally added what is being loaded (hence the `strings` part), since @haqadn pointed out that it's not clear what "Load defaults" loads when using the button when I initially added it to the dashboard in Cornerstone Pages.

Now I'm confused 😅 

I also think we're getting a bit carried away with details.

Comment (haqadn) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I think **handle** is fine considering this terminology is coming WP itself, [Example](https://developer.wordpress.org/reference/functions/wp_enqueue_script/).

Comment (dilirity) [reply projects/plugins/boost/app/assets/src/js/features/minify-meta/minify-meta.tsx]: I've updated the UI accordingly in 04a0dfd.

**Review by haqadn — COMMENTED** [review]
Looks good to me. I did make some changes. Since I am the one who made the latest changes, leaving the approval to others.

**Rationale behind my commit ([b2c5854](https://github.com/Automattic/jetpack/pull/40496/commits/b2c585475b3a5996047408c92d8231c186508bd1)):**
I wanted to keep the stores for CSS and JS separate. This will allow us to easily move the store registrations to individual modules, which we need to do soon.

**Review by dilirity — COMMENTED** [review]
Changes @haqadn did look good.

I think we can :shipit: 

I can't approve, since it's my pr 😅

**Review by donnchawp — APPROVED** [review]
Looks good! :shipit:
